Quantum-Control-Applications/AMO/Use Case 1 - Saffman Lab - Atom Sorting/Array_sorting.py

```diff
@@ -0,0 +1,497 @@
+from qm.qua import *
+from qm.QuantumMachinesManager import QuantumMachinesManager
+from qm import SimulationConfig
+from qm import LoopbackInterface
+from Config_array_sorting import *
+import matplotlib.pyplot as plt
+
+
+################################
+# Open quantum machine manager #
+################################
+qmm = QuantumMachinesManager()
+
+########################
+# Open quantum machine #
+########################
+qm = qmm.open_qm(config)
+
+########################################
+# Define target matrix and frequencies #
+########################################
+piecewise_chirp = False
+analog_occupation_matrix = False
+raw_adc_acquisition = True
+single_run = True
+# Initial occupation matrix in 2D
+atom_location_list = [list(map(int, np.random.choice([0, 1], size=number_of_columns, p=[.4, .6])))
+                      for i in range(number_of_rows)]
+# Initial occupation matrix in 1D
+atom_location_list_1D = [j for sub in atom_location_list for j in sub]
+
+# Target occupation matrix in 2D   TODO fill this
+goal = [
+    ['1', 'X', 'X', 'X', 'X', 'X', '1'],
+    ['X', '1', 'X', 'X', 'X', '1', 'X'],
+    ['X', 'X', '1', '1', '1', 'X', 'X'],
+    ['X', 'X', '1', '1', '1', 'X', 'X'],
+    ['X', 'X', '1', '1', '1', 'X', 'X'],
+    ['X', '1', 'X', 'X', 'X', '1', 'X'],
+    ['1', 'X', 'X', 'X', 'X', 'X', '1']]
+
+atom_target_list = [[1 if goal[j][i] == '1' else 0 for i in range(number_of_columns)] for j in range(number_of_rows)]
+# Target occupation matrix in 1D
+atom_target_list_1d = [j for sub in atom_target_list for j in sub]
+print("Atom target matrix:")
+print_2d(atom_target_list)
+print("Initial occupation matrix:")
+print_2d(atom_location_list)
+
+# Target column frequencies in 1D and of size sum of 1s in target
+target_frequencies_full_python = [[int(column_if[i]) if atom_target_list[j][i] else 0 for i in range(len(atom_target_list[0]))]
+                           for j in range(number_of_rows)]
+target_frequencies_1D = [j for sub in target_frequencies_full_python for j in sub]
+
+#############################################
+# Macros used to derive relevant parameters #
+#############################################
+
+
+def find_number_of_tweezers():

```

Comment:
need to change Macros to match our "best practice guide"
https://qm-docs.qualang.io/guides/best_practices#macros

Reply:
I can do this of course, but I thought that the use-cases should correspond to the program that did run on a specific experiment without modifications as stated in Schluster use-case...

Reply:
hmm, I understand, but I'm not sure we want to postcode that doesn't correspond to our own practices...
I think that the main thing is that use-cases are not meant to be "runnable".

It's not a functional change, not different than changing a variable name.

@tomerfqm WDYT?

Reply:
Oh, I see that you actually already made the change?

Reply:
I understand but to me it is a bit more than just changing a variable name, because before the variables were global and could change from anywhere in the program. Now they are local and so the logic may have changed which will result in a different program... I tested it on Bordeaux lab and it seems to behave properly...
In my opinion there are two ways of presenting the use-cases: either (1) they are codes that ran on real experiment and so it is sure that it works but it is hardware specific and so adaptations need to be made if someone else wants to implement it and we cannot update them if our best practice changed for instance, or (2) they are general (not hardware specific) programs that do not correspond to any experiment and will need adaptations before being implemented, but they will follow our best practice and can be modified if needed.
If think we need to discuss it and choose the most relevant case while we do not have too many use-cases...

Reply:
> Oh, I see that you actually already made the change?

Yes but I saved the two versions

Reply:
Yes, you're right in principle about the macros. The thing is, if they were used as global (changed outside of the macro), then it is very very confusing...
I think that we are going for (1), and we definitely will not update them if we update our best practices. Maybe we can just add a note regarding it...
In this specific case, I would say that the main problem is that at Saffman they were written without the best practice :)

I'm also ok with keeping the macros as they were, and adding a note about it in the readme file, something like:
"Note that the macros are not written according to the best practice and are using global variables, use with care"
whatever you prefer.

Reply:
Ok I understand, I 100% prefer to upload the best practice one with the modified macro because it is indeed much clearer.
By the way, I think I can also add macros for get the locations and frequencies of each row and another one for reading the occupation matrix from the analog input to make the main program more concise

Reply:
This is up to you :) 
From my POV it is ready.

I still haven't looked at the readme too deeply, but we can release and fix it afterward if any changes are needed...